### PR TITLE
Adds p_They (and friends) for capitalized pronoun helpers

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
image

Basically replaces p_they(TRUE) with p_They

## Why It's Good For The Game
Fewer ambiguous parameters (what does TRUE mean when passed to p_they?) More comprehensive helper functions. 

## Changelog
:cl: Tattle
spellcheck: Fixed the grammar on a few revenant messages
/:cl:

### CODE REVIEW

File: code/__HELPERS/pronouns.dm [hunk 1]
```diff
@@ -69,10 +77,11 @@
 			. = "she"
 		if(MALE)
 			. = "he"
-	if(capitalized)
-		. = capitalize(.)
 
-/client/p_their(capitalized, temp_gender)
+/client/p_They(temp_gender)
+	. = capitalize(p_they(temp_gender))
```
Comment (dragomagol) [inline code/__HELPERS/pronouns.dm:82]:
```suggestion
	. = capitalize(p_They(temp_gender))
```

File: code/__HELPERS/pronouns.dm [hunk 2]
```diff
@@ -81,8 +90,9 @@
 			. = "her"
 		if(MALE)
 			. = "his"
-	if(capitalized)
-		. = capitalize(.)
+
+/client/p_Their(temp_gender)
+	. = capitalize(p_their(temp_gender))
```
Comment (LemonInTheDark) [inline code/__HELPERS/pronouns.dm:95]: isn't this redundant?

File: code/__HELPERS/pronouns.dm [hunk 3]
```diff
@@ -174,8 +185,9 @@
 			. = "his"
 		if(PLURAL)
 			. = "their"
-	if(capitalized)
-		. = capitalize(.)
+
+/mob/p_Their(temp_gender)
+	. = capitalize(p_their(temp_gender))
```
Comment (LemonInTheDark) [inline code/__HELPERS/pronouns.dm:190]: similar for most of these

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/__HELPERS/pronouns.dm [hunk 4]
```diff
@@ -32,11 +34,17 @@
...
+/datum/proc/p_Theyve(temp_gender)
+	. = p_They(temp_gender) + "'" + copytext_char(p_have(temp_gender), 3)
+
+/datum/proc/p_theyre(temp_gender)
+	. = p_they(temp_gender) + "'" + copytext_char(p_are(temp_gender), 2)
 
-/datum/proc/p_theyre(capitalized, temp_gender)
-	. = p_they(capitalized, temp_gender) + "'" + copytext_char(p_are(temp_gender), 2)
+/datum/proc/p_Theyre(temp_gender)
+	. = p_They(temp_gender) + "'" + copytext_char(p_are(temp_gender), 2)
```
Comment (Mothblocks) [inline code/__HELPERS/pronouns.dm:47]: Let's make all these just `return`

**Review by Mothblocks — COMMENTED** [review]
Out of curiosity, does this change every consumer? There's not as many changes as I expected

File: code/__HELPERS/pronouns.dm [hunk 5]
```diff
@@ -1,259 +1,259 @@
...
 	switch(temp_gender)
 		if(FEMALE)
-			. = "hers"
+			return "hers"
 		if(MALE)
-			. = "his"
+			return "his"
 	if(capitalized)
-		. = capitalize(.)
+		return capitalize(.)
```
Comment (LemonInTheDark) [inline code/__HELPERS/pronouns.dm:101]: why retain this here?

File: code/__HELPERS/pronouns.dm [hunk 6]
```diff
@@ -1,259 +1,259 @@
...
 		temp_gender = gender
-	. = "theirs"
+	return "theirs"
 	switch(temp_gender)
 		if(FEMALE)
-			. = "hers"
+			return "hers"
 		if(MALE)
-			. = "his"
+			return "his"
```
Comment (LemonInTheDark) [inline code/__HELPERS/pronouns.dm:99]: regex strikes again (else case for the switch would work best I think, or just stick it below)

**Comment by dragomagol** [comment]
image

This should change every consumer! I didn't expect this few changes either, but I guess p_they() is just more common.

File: code/__HELPERS/pronouns.dm [hunk 7]
```diff
@@ -1,259 +1,259 @@
...
 	if(capitalized)
-		. = capitalize(.)
+		return capitalize(.)
 
 /datum/proc/p_them(capitalized, temp_gender)
-	. = "it"
+	return "it"
 	if(capitalized)
-		. = capitalize(.)
+		return capitalize(.)
```
Comment (LemonInTheDark) [inline code/__HELPERS/pronouns.dm:23]: why retail the arg? legacy support? (also regex again etc etc)

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/__HELPERS/pronouns.dm (hunk 5 above)
Comment (dragomagol) [reply code/__HELPERS/pronouns.dm:101]: Forgot, most likely

File: code/__HELPERS/pronouns.dm (hunk 6 above)
Comment (dragomagol) [reply code/__HELPERS/pronouns.dm:99]: Fixing it now!

**Comment by LemonInTheDark** [comment]
winning

**Pull request merged**